mesh/mesh.cpp

```diff
@@ -1833,11 +1854,12 @@ void Mesh::SetAttributes()
       MFEM_WARNING("Non-positive attributes on the boundary!");
    }
 
-   attribs.SetSize(GetNE());
-   for (int i = 0; i < attribs.Size(); i++)
-   {
-      attribs[i] = GetAttribute(i);
-   }
+   // assume that attributes have been modified
+   ElementAttributesUpdated();
+
+   // since we're already reading all the attributes, might as well just update the cache
+   attribs = GetElementAttributes();
+
    attribs.Sort();
    attribs.Unique();
    attribs.Copy(attributes);
```

Comment:
Not really important, but I wonder if we should use `std::unordered_set` or similar here.